.github/workflows/publish.yml

```diff
@@ -0,0 +1,48 @@
+name: GitHub action publish new version
+on:
+    workflow_dispatch:
+        inputs:
+            versionType:
+                type: choice
+                description: Version Type
+                required: true
+                options:
+                    - patch
+                    - minor
+                    - major
+permissions:
+    contents: write
+jobs:
+    publish:
+        name: Publish github action version
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout repository
+              uses: actions/checkout@v4
+
+            - name: Use Node.js 20
+              uses: actions/setup-node@v3
+              with:
+                node-version: 20.x
+
+            - name: Install npm deps
+              run: npm ci
+
+            - name: Configure git
+              run: git config --global user.email "y-infra@yandex.ru" && git config --global user.name "y-infra"
+
+            - name: Bump version
+              run: npm version ${{ github.event.inputs.versionType }}
+
+            - name: Build github action
+              run: npm run ci:build
+
+            - name: Commit version
+              run: |
+                git add dist -f && \
+                git commit --amend --no-edit dist && \
+                TAG=`git describe --tags $(git rev-list --tags --max-count=1)` && \
+                git tag -d $TAG && \
+                git tag $TAG && \
```

Comment:
After "git commit --amend" we need to move release tag, because it still references old commit hash